source/common/upstream/load_balancer_impl.cc

```diff
@@ -1084,28 +1084,52 @@ void EdfLoadBalancerBase::refresh(uint32_t priority) {
       // Skip edf creation.
       return;
     }
-    scheduler.edf_ = std::make_unique<EdfScheduler<const Host>>();
-
-    // Populate scheduler with host list.
-    // TODO(mattklein123): We must build the EDF schedule even if all of the hosts are currently
-    // weighted 1. This is because currently we don't refresh host sets if only weights change.
-    // We should probably change this to refresh at all times. See the comment in
-    // BaseDynamicClusterImpl::updateDynamicHostList about this.
-    for (const auto& host : hosts) {
-      // We use a fixed weight here. While the weight may change without
-      // notification, this will only be stale until this host is next picked,
-      // at which point it is reinserted into the EdfScheduler with its new
-      // weight in chooseHost().
-      scheduler.edf_->add(hostWeight(*host), host);
+
+    // If there are no hosts or a single one, there is no need for an EDF scheduler
```

Comment:
Nit: probably safe but not runtime guarded.

Reply:
Better be safe than sorry. Done.